src/bin/pgcopydb/follow.c

```diff
@@ -321,8 +319,9 @@ follow_main_loop(CopyDataSpec *copySpecs, StreamSpecs *streamSpecs)
 		}
 
 		/* switch to the next mode, increment loop counter */
-		previousMode = currentMode;
-		currentMode = modeArray[++loop % count];
+		size_t prev = loop % count;
+		LogicalStreamMode previousMode = modeArray[prev];
+		LogicalStreamMode currentMode = modeArray[prev + 1];
```

Comment:
On a quick review I can't see how that doesn't change the ordering of the selected modes. What's the first being used? it needs to be STREAM_MODE_CATCHUP and it needs to be easy to assert that when a human reads the code. I do feel like we are not keeping that property here (it's confusing now).

Reply:
In the original version of the code the assignment `LogicalStreamMode previousMode = STREAM_MODE_UNKNOW;` is overriden with `previousMode = currentMode;` without using the value, thus it is redundant.